webapi/Services/AzureContentModerator.cs

```diff
@@ -0,0 +1,157 @@
+
+        this._httpClientHandler = new() { CheckCertificateRevocationList = true };
+        this._httpClient = new(this._httpClientHandler);
+
+        this._httpClient.DefaultRequestHeaders.Add("User-Agent", HttpUserAgent);
+
+        // Subscription Key header required to authenticate requests to Azure API Management (APIM) service
+        this._httpClient.DefaultRequestHeaders.Add("Ocp-Apim-Subscription-Key", apiKey);
+    }
+
+    /// <summary>
+    /// Parse the analysis result and return the violated categories.
+    /// </summary>
+    /// <param name="analysisResult">The content analysis result.</param>
+    /// <param name="threshold">The violation threshold.</param>
+    /// <returns>The list of violated category names. Will return an empty list if there is no violation.</returns>
+    public static List<string> ParseViolatedCategories(Dictionary<string, AnalysisResult> analysisResult, short threshold)
+    {
+        var violatedCategories = new List<string>();
+
+        foreach (var category in analysisResult.Values)
+        {
+            if (category.RiskLevel > threshold)
+            {
+                violatedCategories.Add(category.Category);
+            }
+        }
+        return violatedCategories;
+    }
+
+    /// <summary>
+    /// Invokes a sync API to perform harmful content analysis on image.
+    /// <param name="base64Image">Base64 envoding content of image</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// </summary>
+    /// <returns>SKContext containing the image analysis result.</returns>
+    public async Task<Dictionary<string, AnalysisResult>> ImageAnalysisAsync(string base64Image, CancellationToken cancellationToken)
+    {
+        var image = base64Image.Replace("data:image/png;base64,", "", StringComparison.InvariantCultureIgnoreCase).Replace("data:image/jpeg;base64,", "", StringComparison.InvariantCultureIgnoreCase);
+        ImageContent content = new(image);
+        ImageAnalysisRequest requestBody = new(content, s_categories);
+
+        using var httpRequestMessage = new HttpRequestMessage()
+        {
+            Method = HttpMethod.Post,
+            RequestUri = new Uri($"{this._endpoint}/contentsafety/image:analyze?api-version=2023-04-30-preview"),
+            Content = new StringContent(JsonSerializer.Serialize(requestBody), Encoding.UTF8, "application/json"),
+        };
+
+        var response = await this._httpClient.SendAsync(httpRequestMessage, cancellationToken).ConfigureAwait(false);
+        var body = await response.Content.ReadAsStringAsync(cancellationToken).ConfigureAwait(false);
+        if (!response.IsSuccessStatusCode || body is null)
+        {
+            throw new AIException(
+                AIException.ErrorCodes.UnknownError,
+                $"Content moderator: Failed analyzing the image. {response.StatusCode}");
+        }
+
+        var result = JsonSerializer.Deserialize<Dictionary<string, AnalysisResult>>(body!);
```

Comment:
Can we create a model to parse this into rather than a dictionary?

Reply:
Done - see ImageAnalysisResponse